pandas/core/arrays/string_.py

```diff
@@ -122,6 +122,9 @@ class StringArray(PandasArray):
 
     copy : bool, default False
         Whether to copy the array of data.
+    convert : bool, default False
```

Comment:
we don't want this parameter generally in the actual constructors (you can instead do this in `_from_sequence), but even there we don't do coercion. I think this actually needs to be done in the caller. cc @jorisvandenbossche @TomAugspurger

Reply:
> we don't want this parameter generally in the actual constructors (you can instead do this in `_from_sequence)

The decision to validate or convert needs to be in the `__init__`, because the validation already is there and it has no way to know from the input array, whether the input array only contains strings and NA's. So it needs a flag to tell it if it should validate or convert because we don't want to do both.

An alternative is to always run the conversion step. In that case validation is not needed. It would not cost in performance, after #36162 has been merged: 

```python
>>> x = np.array([str(u) for u in range(1_000_000)], dtype=object)
>>> %timeit pd._libs.lib.ensure_string_array(x, copy=False)
11.5 ms ± 145 µs per loop
>>> %timeit pd._libs.lib.is_string_array(x)
13.3 ms ± 80.5 µs per loop
```

Reply:
I seem to recall we had discussion about trying to avoid this double validation, but don't directly find anything on the original PR https://github.com/pandas-dev/pandas/pull/27949

> So it needs a flag to tell it if it should validate or convert because validation after conversion is wasteful.

Alternative would be a flag to disable validation (instead of a flag to enable conversion),then `_from_sequence` can do the conversion and signal to the constructor it doesn't need to validate.

In general, I prefer the Array class constructors to be minimal and performant (they are not meant for say-to-day users, those should use `pd.array(..)`), eg the IntegerArray constructor only checks that the passed values are actually an ndarray and have the correct dtype (and not any other coercion or conversion). But of course, there it is a relatively cheap check, while for object dtype array this check is expensive. 
So given that limiation for StringArray using object dtype, I am fine with adding such a "fastpath" keyword in this case (but I would use something like `validate` or `verify_integrity`)